server/src/main/java/org/eclipse/lsp/cobol/service/delegates/completions/SectionCompletion.java

```diff
@@ -14,41 +14,42 @@
  */
 package org.eclipse.lsp.cobol.service.delegates.completions;
 
-import org.eclipse.lsp.cobol.service.CobolDocumentModel;
-import org.eclipse.lsp.cobol.service.delegates.validations.AnalysisResult;
 import com.google.inject.Singleton;
 import lombok.NonNull;
+import org.eclipse.lsp.cobol.service.CobolDocumentModel;
+import org.eclipse.lsp4j.CompletionItem;
 import org.eclipse.lsp4j.CompletionItemKind;
 
+import javax.annotation.Nullable;
 import java.util.Collection;
-import java.util.Collections;
-import java.util.Optional;
+import java.util.function.Function;
 
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
 import static org.eclipse.lsp.cobol.service.delegates.completions.CompletionOrder.SECTIONS;
 
-/**
- * Provide completion functionality for sections
- */
+/** Provides completion functionality for sections */
 @Singleton
 public class SectionCompletion implements Completion {
 
   @Override
-  public Collection<String> getCompletionSource(CobolDocumentModel document) {
-    return Optional.ofNullable(document)
-        .map(CobolDocumentModel::getAnalysisResult)
-        .map(AnalysisResult::getSections)
-        .orElse(Collections.emptySet());
+  public @NonNull Collection<CompletionItem> getCompletionItems(
+      @NonNull String token, @Nullable CobolDocumentModel document) {
+    if (document == null) return emptyList();
+    return document.getAnalysisResult().getSections().stream()
+        .filter(CompletionUtils.startsWithIgnoreCase(token))
+        .map(toSectionCompletion())
+        .collect(toList());
   }
 
-  @NonNull
-  @Override
-  public String getSortOrderPrefix() {
-    return SECTIONS.prefix;
-  }
-
-  @NonNull
-  @Override
-  public CompletionItemKind getKind() {
-    return CompletionItemKind.Function;
+  private Function<String, CompletionItem> toSectionCompletion() {
+    return name -> {
```

Comment:
Optional: I guess, is it possible to extract this as a  method, then one "return" can removed (implicit lambda return)